etl/src/main/java/org/pih/hivmigration/etl/sql/SqlMigrator.java

```diff
@@ -317,4 +321,68 @@ private String abbreviate(String str) {
         }
     }
 
+    // Methods that allow for creating tests that should pass
+
+    protected void assertAllRows(String description, String mysqlQuery) throws Exception {
+        log.info("Testing: " + description);
+        Number allRows = (Number) selectMysql("select count(*) from drug_order", new ScalarHandler<Number>());
+        Number result = (Number) selectMysql(mysqlQuery, new ScalarHandler<Number>());
+        if (!allRows.equals(result)) {
+            fail(allRows + " expected but " + result + " returned");
+        }
+        log.info("Test successful");
+    }
+
+    protected void assertNoRows(String description, String mysqlQuery) throws Exception {
+        log.info("Testing: " + description);
+        Number result = (Number) selectMysql(mysqlQuery, new ScalarHandler<Number>());
+        if (result.intValue() != 0) {
+            fail("0 rows expected but " + result + " returned");
+        }
+        log.info("Test successful");
+    }
+
+    protected void assertMatch(String description, String oracleQuery, String mysqlQuery) throws Exception {
+        log.info("Testing: " + description);
+        List<Map<String, Object>> oracleResults = (List<Map<String, Object>>) selectOracle(oracleQuery, new MapListHandler());
+        List<Map<String, Object>> mysqlResults = (List<Map<String, Object>>) selectMysql(mysqlQuery, new MapListHandler());
+        if (oracleResults.size() != mysqlResults.size()) {
+            fail("Oracle has " + oracleResults.size() + " rows, MySQL has: " + mysqlResults.size());
+        }
+        for (int i=0; i<oracleResults.size(); i++) {
+            Map<String, Object> oracleRow = oracleResults.get(i);
+            Map<String, Object> mysqlRow = mysqlResults.get(i);
+            if (oracleRow.size() != mysqlRow.size()) {
+                fail("Oracle has " + oracleRow.size() + " columns, MySQL has: " + mysqlRow.size());
+            }
+            Set<String> allKeys = new HashSet<>();
+            for (String key : oracleRow.keySet()) {
+                allKeys.add(key.toLowerCase());
+            }
+            for (String key : mysqlRow.keySet()) {
+                allKeys.add(key.toLowerCase());
+            }
+            for (String key : allKeys) {
+                Object oracleValue = oracleRow.get(key.toUpperCase());
+                Object mysqlValue = mysqlRow.get(key);
+                if (oracleValue != null || mysqlValue != null) {
```

Comment:
Can remove this outer `if` block and just make that last `else` into an `else if(mysqlValue == null && oracleValue == null)`.